packages/ddp-server/livedata_server.js

```diff
@@ -809,24 +809,33 @@ _.extend(Session.prototype, {
     self.collectionViews = {};
     self.userId = userId;
 
-    // Save the old named subs, and reset to having no subscriptions.
-    var oldNamedSubs = self._namedSubs;
-    self._namedSubs = {};
-    self._universalSubs = [];
+    // _setUserId is normally called from a Meteor method with
+    // DDP._CurrentInvocation set. But DDP._CurrentInvocation is not expected
+    // to be set inside a publish function, so we temporary unset it.
+    // Otherwise, this can lead to a problem that if a publish function is
```

Comment:
I would remove the part of the comment from here on, because it is not anymore true. `connection` will now be always set in server-side methods, but we do not want `DDP._CurrentInvocation` to be set for those who are accessing it directly to check if they are inside a method call.